pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -64,21 +67,28 @@ func NewScheduler(nodeTemplates []*scheduling.NodeTemplate, cluster *state.Clust
 			return true
 		}
 		s.inflight = append(s.inflight, NewInFlightNode(node, s.topology, nodeTemplate.StartupTaints, s.daemonOverhead[nodeTemplate]))
+
+		// We don't use the status field and instead recompute the remaining resources to ensure we have a consistent view
```

Comment:
Why don't we use the status field?

Reply:
It's not guaranteed to be updated yet.  At least this way we have a consistent view of the cluster when scheduling (or as consistent as we can be).  The other benefit is that this will continue working as I expect when we stop creating nodes that `ForEachNode` will still iterate over nodes that are in the process of being launched, meaning this will continue to work.

Reply:
So IIUC, in a followon PR you'll update the limit in the provisioning loop and then replace `remainingResources` using provisioner.status.limits?

Reply:
At some point, and likely in the same PR that I fix our metrics code scale problems.